### ISSUE TITLE
ZTC ZT230-203dpi ZPL PDF Printer Label Move Down about 1/4" inch

### ISSUE THREAD

**Issue by cupsman** [OpenPrinting/cups#515]
Hi Michael and team,

Questions:
How can I make this label print correctly using the proper driver settings?  How do I move the label down about 1/4" or so?
Do you have to restart cups if you manually update the ppd file (aka - ImageableArea)
Is moving to 6x4 in CUPS 1.6.3 the proper way to make jobs automatically landscape?

There's a few things I can't figure out.  
When I use 4x6 as the selected page size it won't print landscape.  If I use 6x4 it prints landscape automatically.

I can see the file comes into a pdf format with this page size:  (pdfinfo shows: Page size:      326.05 x 85.05 pts.  When it's in this format I have to use -o landscape to get it to rotate and then it's about 1/4 inch to high on the label.  This is how the print job is coming and I can't manipulate the file.

I did see in some other google searches I found some similar scenarios.  I took the printjob sent it through a postscript driver, and then ps2pdf (Manually) and it created the page size as:  85.05 x 326.05.  With this format it prints perfectly without any options and it formats it on the label lined up correctly.

*DefaultPageSize: w432h288
*PageSize w432h288/6.00x4.00": "<</PageSize[432 288]/ImagingBBox null>>setpagedevice"
*DefaultPageRegion: w432h288
*PageRegion w432h288/6.00x4.00": "<</PageSize[432 288]/ImagingBBox null>>setpagedevice"
*DefaultImageableArea: w432h288

*ImageableArea w432h288/6.00x4.00": "36 36 432 288"
*DefaultPaperDimension: w432h288
*PaperDimension w432h288/6.00x4.00": "432 288"

 *cupsFilter: "application/vnd.cups-raster 0 rastertolabel"

Using this Driver File:
*PPD-Adobe: "4.3"
*%%%% PPD file for ZPL Label Printer with CUPS.
*%%%% Created by the CUPS PPD Compiler CUPS v1.6.3.

**Comment by zdohnal** [comment]
Hi @cupsman ,

it would be great if you attached CUPS debug logs, pdf label you want to print and your PPD you use for the printer - adding .txt suffix to any file or using zip to get an archive should make the file visible for github and you can attach it as a file.

Once I have the necessary data, I will try to reproduce the issue with my Zebra printer on the latest CUPS version, but I don't have the same model, so it will be mostly sanity check.

Your best course of action would be to contact your OS support, since version 1.6.3 is about a decade old and we in OpenPrinting/CUPS mostly focus on the latest released version. Or upgrade to the latest CUPS and cups-filters released versions - 2.4.2 and 1.28.16.

**Comment by cupsman** [comment]
Can you remove the two attachments please.

**Comment by michaelrsweet** [comment]
@cupsman Done!

**Comment by cupsman** [comment]
Here's the error_log file for the job.
The bottom line on this is the rasterimage is getting cutoff at the top like by a 1/4" or so.  Alot of good information for troubleshooting at the top of the issue.

I [01/Nov/2022:18:37:34 +0000] [Job 206] Adding start banner page "none".
I [01/Nov/2022:18:37:34 +0000] [Job 206] Queued on "PRINTERNAME" by "myuserid".
D [01/Nov/2022:18:37:34 +0000] [Job 206] Auto-typing file...
D [01/Nov/2022:18:37:34 +0000] [Job 206] Request file type is application/pdf.
I [01/Nov/2022:18:37:34 +0000] [Job 206] File of type application/pdf queued by "myuserid".
I [01/Nov/2022:18:37:34 +0000] [Job 206] Adding end banner page "none".
D [01/Nov/2022:18:37:34 +0000] [Job 206] time-at-processing=1667327854
D [01/Nov/2022:18:37:34 +0000] [Job 206] 3 filters for job:
D [01/Nov/2022:18:37:34 +0000] [Job 206] pdftopdf (application/pdf to application/vnd.cups-pdf, cost 66)
D [01/Nov/2022:18:37:34 +0000] [Job 206] gstoraster (application/vnd.cups-pdf to application/vnd.cups-raster, cost 99)
D [01/Nov/2022:18:37:34 +0000] [Job 206] rastertolabel (application/vnd.cups-raster to printer/PRINTERNAME, cost 0)
D [01/Nov/2022:18:37:34 +0000] [Job 206] job-sheets=none,none
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[0]="PRINTERNAME"
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[1]="206"
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[2]="myuserid"
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[3]="test"
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[4]="1"
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[5]="finishings=3 number-up=1 orientation-requested=6 job-uuid=urn:uuid:b139eeff-41a7-3d76-7e57-76e8eca32c10 job-originating-host-name=localhost time-at-creation=1667327854 time-at-processing=1667327854"
D [01/Nov/2022:18:37:34 +0000] [Job 206] argv[6]="/my_cups/spool/d00206-001"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[0]="CUPS_CACHEDIR=/var/cache/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[1]="CUPS_DATADIR=/usr/share/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[2]="CUPS_DOCmyuserid=/usr/share/cups/www"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[3]="CUPS_FONTPATH=/usr/share/cups/fonts"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[4]="CUPS_REQUESTmyuserid=/my_cups/spool"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[5]="CUPS_SERVERBIN=/usr/lib/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[6]="CUPS_SERVERmyuserid=/etc/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[7]="CUPS_STATEDIR=/var/run/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[8]="HOME=/my_cups/tmp"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[9]="PATH=/usr/lib/cups/filter:/usr/lib64/cups/filter:/usr/bin:/usr/sbin:/bin:/usr/bin"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[10]="SERVER_ADMIN=myemail@address"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[11]="SOFTWARE=CUPS/1.6.3"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[12]="TMPDIR=/my_cups/tmp"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[13]="USER=myuserid"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[14]="CUPS_MAX_MESSAGE=2047"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[15]="CUPS_SERVER=/var/run/cups/cups.sock"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[16]="CUPS_ENCRYPTION=IfRequested"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[17]="IPP_PORT=631"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[18]="CHARSET=utf-8"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[19]="LANG=en_US.UTF-8"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[20]="PPD=/etc/cups/ppd/PRINTERNAME.ppd"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[21]="RIP_MAX_CACHE=128m"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[22]="CONTENT_TYPE=application/pdf"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[23]="DEVICE_URI=socket://XX.XX.XX.XX:9100"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[24]="PRINTER_INFO=PRINTERNAME"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[25]="PRINTER_LOCATION=MYLOCATION"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[26]="PRINTER=PRINTERNAME"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[27]="PRINTER_STATE_REASONS=none"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[28]="CUPS_FILETYPE=document"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[29]="FINAL_CONTENT_TYPE=application/vnd.cups-raster"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[30]="AUTH_I****"
I [01/Nov/2022:18:37:34 +0000] [Job 206] Started filter /usr/lib/cups/filter/pdftopdf (PID 14205)
I [01/Nov/2022:18:37:34 +0000] [Job 206] Started filter /usr/lib/cups/filter/gstoraster (PID 14206)
I [01/Nov/2022:18:37:34 +0000] [Job 206] Started filter /usr/lib/cups/filter/rastertolabel (PID 14207)
I [01/Nov/2022:18:37:34 +0000] [Job 206] Started backend /usr/lib/cups/backend/socket (PID 14208)
D [01/Nov/2022:18:37:34 +0000] [Job 206] STATE: +connecting-to-device
D [01/Nov/2022:18:37:34 +0000] [Job 206] Looking up "XX.XX.XX.XX"...
D [01/Nov/2022:18:37:34 +0000] [Job 206] PID 14205 (/usr/lib/cups/filter/pdftopdf) exited with no errors.
D [01/Nov/2022:18:37:34 +0000] [Job 206] PPD uses qualifier 'Gray.Saved.200dpi'
D [01/Nov/2022:18:37:34 +0000] [Job 206] Calling FindDeviceById(cups-PRINTERNAME)
D [01/Nov/2022:18:37:34 +0000] [Job 206] Found device /org/freedesktop/ColorManager/devices/cups_PRINTERNAME
D [01/Nov/2022:18:37:34 +0000] [Job 206] Calling GetProfileForQualifiers(Gray.Saved.200dpi...)
D [01/Nov/2022:18:37:34 +0000] [Job 206] Found profile /org/freedesktop/ColorManager/profiles/PRINTERNAME_Gray__
D [01/Nov/2022:18:37:34 +0000] [Job 206] Calling org.freedesktop.ColorManager.Profile.Get(Filename)
D [01/Nov/2022:18:37:34 +0000] [Job 206] Use profile filename: ''
D [01/Nov/2022:18:37:34 +0000] [Job 206] Using ICC Profile ''
D [01/Nov/2022:18:37:34 +0000] [Job 206] Ghostscript command line: /usr/bin/gs -dQUIET -dPARANOIDSAFER -dNOPAUSE -dBATCH -dNOINTERPOLATE -sDEVICE=cups -sstdout=%stderr -sOutputFile=%stdout -dAdvanceDistance=1000 -r200x200 -dDEVICEWIDTHPOINTS=432 -dDEVICEHEIGHTPOINTS=288 -dcupsBitsPerColor=1 -dcupsColorOrder=0 -dcupsColorSpace=3 -dcupsCompression=-1 -dcupsRowStep=200 -scupsPageSizeName=w432h288 -I/usr/share/cups/fonts -c -f -_
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[0]="CUPS_CACHEDIR=/var/cache/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[1]="CUPS_DATADIR=/usr/share/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[2]="CUPS_DOCmyuserid=/usr/share/cups/www"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[3]="CUPS_FONTPATH=/usr/share/cups/fonts"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[4]="CUPS_REQUESTmyuserid=/my_cups/spool"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[5]="CUPS_SERVERBIN=/usr/lib/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[6]="CUPS_SERVERmyuserid=/etc/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[7]="CUPS_STATEDIR=/var/run/cups"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[8]="HOME=/my_cups/tmp"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[9]="PATH=/usr/lib/cups/filter:/usr/lib64/cups/filter:/usr/bin:/usr/sbin:/bin:/usr/bin"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[10]="SERVER_ADMIN=myemail@address"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[11]="SOFTWARE=CUPS/1.6.3"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[12]="TMPDIR=/my_cups/tmp"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[13]="USER=myuserid"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[14]="CUPS_MAX_MESSAGE=2047"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[15]="CUPS_SERVER=/var/run/cups/cups.sock"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[16]="CUPS_ENCRYPTION=IfRequested"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[17]="IPP_PORT=631"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[18]="CHARSET=utf-8"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[19]="LANG=en_US.UTF-8"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[20]="PPD=/etc/cups/ppd/PRINTERNAME.ppd"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[21]="RIP_MAX_CACHE=128m"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[22]="CONTENT_TYPE=application/pdf"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[23]="DEVICE_URI=socket://XX.XX.XX.XX0:9100"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[24]="PRINTER_INFO=PRINTERNAME"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[25]="PRINTER_LOCATION=MYLOCATION"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[26]="PRINTER=PRINTERNAME"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[27]="PRINTER_STATE_REASONS=none"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[28]="CUPS_FILETYPE=document"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[29]="FINAL_CONTENT_TYPE=application/vnd.cups-raster"
D [01/Nov/2022:18:37:34 +0000] [Job 206] envp[30]="AUTH_INFO_REQUIRED=none"
I [01/Nov/2022:18:37:34 +0000] [Job 206] Start rendering...
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Start rendering...", current level=INFO
I [01/Nov/2022:18:37:34 +0000] [Job 206] Processing page 1...
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Processing page 1...", current level=INFO
I [01/Nov/2022:18:37:34 +0000] [Job 206] Processing page 2...
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Processing page 2...", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] PAGE: 1 1
I [01/Nov/2022:18:37:34 +0000] [Job 206] Starting page 1.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Starting page 1.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] StartPage...
D [01/Nov/2022:18:37:34 +0000] [Job 206] MediaClass = ""
D [01/Nov/2022:18:37:34 +0000] [Job 206] MediaColor = ""
D [01/Nov/2022:18:37:34 +0000] [Job 206] MediaType = ""
D [01/Nov/2022:18:37:34 +0000] [Job 206] OutputType = ""
D [01/Nov/2022:18:37:34 +0000] [Job 206] AdvanceDistance = 1000
D [01/Nov/2022:18:37:34 +0000] [Job 206] AdvanceMedia = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] Collate = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] CutMedia = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] Duplex = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] HWResolution = [ 200 200 ]
D [01/Nov/2022:18:37:34 +0000] [Job 206] ImagingBoundingBox = [ 0 0 326 85 ]
D [01/Nov/2022:18:37:34 +0000] [Job 206] InsertSheet = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] Jog = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] LeadingEdge = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] Margins = [ 0 0 ]
D [01/Nov/2022:18:37:34 +0000] [Job 206] ManualFeed = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] MediaPosition = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] MediaWeight = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] MirrorPrint = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] NegativePrint = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] NumCopies = 1
D [01/Nov/2022:18:37:34 +0000] [Job 206] Orientation = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] OutputFaceUp = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] PageSize = [ 326 85 ]
D [01/Nov/2022:18:37:34 +0000] [Job 206] Separations = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] TraySwitch = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] Tumble = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsWidth = 906
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsHeight = 236
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsMediaType = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsBitsPerColor = 1
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsBitsPerPixel = 1
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsBytesPerLine = 114
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsColorOrder = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsColorSpace = 3
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsCompression = -1
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsRowCount = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsRowFeed = 0
D [01/Nov/2022:18:37:34 +0000] [Job 206] cupsRowStep = 200
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 0% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 0% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=0
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 6% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 6% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=6
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 13% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 13% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=13
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 20% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 20% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=20
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 27% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 27% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=27
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 33% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 33% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=33
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 40% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 40% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=40
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 47% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 47% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=47
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 54% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 54% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=54
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 61% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 61% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=61
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 67% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 67% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=67
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 74% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 74% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=74
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 81% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 81% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=81
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 88% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 88% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=88
I [01/Nov/2022:18:37:34 +0000] [Job 206] Printing page 1, 94% complete.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Printing page 1, 94% complete.", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] ATTR: job-media-progress=94
I [01/Nov/2022:18:37:34 +0000] [Job 206] Finished page 1.
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Finished page 1.", current level=INFO
I [01/Nov/2022:18:37:34 +0000] [Job 206] Rendering completed
D [01/Nov/2022:18:37:34 +0000] [Job 206] Set job-printer-state-message to "Rendering completed", current level=INFO
D [01/Nov/2022:18:37:34 +0000] [Job 206] PID 14206 (/usr/lib/cups/filter/gstoraster) exited with no errors.
D [01/Nov/2022:18:37:34 +0000] [Job 206] PID 14207 (/usr/lib/cups/filter/rastertolabel) exited with no errors.
D [01/Nov/2022:18:37:38 +0000] [Job 206] prtGeneralCurrentLocalization type is 0, expected 2!
D [01/Nov/2022:18:37:38 +0000] [Job 206] backendWaitLoop(snmp_fd=6, addr=0x55c949f7caa8, side_cb=0x55c948130380)
D [01/Nov/2022:18:37:38 +0000] [Job 206] Connecting to XX.XX.XX.XX0:9100
I [01/Nov/2022:18:37:38 +0000] [Job 206] Connecting to printer.
D [01/Nov/2022:18:37:38 +0000] [Job 206] Set job-printer-state-message to "Connecting to printer.", current level=INFO
D [01/Nov/2022:18:37:38 +0000] [Job 206] STATE: -connecting-to-device
I [01/Nov/2022:18:37:38 +0000] [Job 206] Connected to printer.
D [01/Nov/2022:18:37:38 +0000] [Job 206] Set job-printer-state-message to "Connected to printer.", current level=INFO
D [01/Nov/2022:18:37:38 +0000] [Job 206] Connected to XX.XX.XX.XX:9100...
D [01/Nov/2022:18:37:38 +0000] [Job 206] backendRunLoop(print_fd=0, device_fd=7, snmp_fd=6, addr=0x55c949f7caa8, use_bc=1, side_cb=0x55c948130380)
D [01/Nov/2022:18:37:38 +0000] [Job 206] Read 8192 bytes of print data...
D [01/Nov/2022:18:37:38 +0000] [Job 206] Wrote 8192 bytes of print data...
D [01/Nov/2022:18:37:38 +0000] [Job 206] Read 8192 bytes of print data...
D [01/Nov/2022:18:37:38 +0000] [Job 206] Wrote 8192 bytes of print data...
D [01/Nov/2022:18:37:38 +0000] [Job 206] Read 2278 bytes of print data...
D [01/Nov/2022:18:37:38 +0000] [Job 206] Wrote 2278 bytes of print data...
I [01/Nov/2022:18:37:38 +0000] [Job 206] Waiting for printer to finish.
D [01/Nov/2022:18:37:38 +0000] [Job 206] Set job-printer-state-message to "Waiting for printer to finish.", current level=INFO
D [01/Nov/2022:18:37:38 +0000] [Job 206] PID 14208 (/usr/lib/cups/backend/socket) exited with no errors.
D [01/Nov/2022:18:37:38 +0000] [Job 206] time-at-completed=1667327858
I [01/Nov/2022:18:37:38 +0000] [Job 206] Job completed.
D [01/Nov/2022:18:37:39 +0000] [Job 206] Unloading...
D [01/Nov/2022:20:02:11 +0000] [Job 206] Loading attributes...
D [01/Nov/2022:20:03:14 +0000] [Job 206] Unloading...

**Comment by zdohnal** [comment]
@cupsman would you mind attaching the file you want to print and PPD file? I don't have cups 1.6.3 at hand.

**Comment by cupsman** [comment]
[PrinterDriver.txt](https://github.com/OpenPrinting/cups/files/9922787/PrinterDriver.txt)
This is the print driver sorry I forgot to close it with comment.

Also (Very important).  I found this the other day and performed it and the label printed perfectly.  Can we chain our filters to match what we did manually here?  This put the label perfectly on the Zebra printer when I manually ran it through the process below and send the filtered.pdf to the printer.

~> /usr/sbin/cupsfilter -P test.ppd -m application/vnd.cups-postscript label.pdf  > filtered.ps
~> ps2pdf filtered.ps
~> pdfinfo filtered.pdf
